src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvert.java

```diff
@@ -71,26 +77,70 @@ private static Path startApartments() {
    */
   @SuppressWarnings("serial")
   public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
+    JsonbConfig config = new JsonbConfig().withAdapters(JsonConvert.getAdapter());
+    Jsonb jsonb = JsonbBuilder.create(config);
```

Comment:
jsonb must be in a try-with-resources. See `apartmentsToJsonString(List<Apartment> listApartments)` function.

Reply:
Fix in f2b1f146000c7e0028b3518608dd590dfd693554